src/server.rs

```diff
@@ -475,11 +475,26 @@ pub trait Output {
     }
 
     fn notify(&self, message: &str) {
-        let output = serde_json::to_string(
-            &NotificationMessage::new(message.to_owned(), ())
-        ).unwrap();
+        let output = serde_json::to_string(&NotificationMessage::new(message.to_owned(), ()))
+            .unwrap();
         self.response(output);
     }
+
+    fn notify_begin(&self) {
+        self.notify("rustDocument/diagnosticsBegin");
```

Comment:
I think you are doing the right thing by factoring out these notification messages, however, I think this is the wrong place for them - the `Output` trait exists at the level of the LS protocol only, and doesn't know much about how the RLS uses it or about the RLS's custom protocol stuff (`rustDocument/...`). I think a better place for these would be somewhere in the `actions` module.